### PR TITLE
fix(man.lua): use `env` command

### PR DESCRIPTION
**Pull request by euclidianAce**
Previously man.lua would use the `env` field in the parameters of `vim.loop.spawn` to override things like MANPAGER. This caused issues on NixOS since `spawn` will _override_ the environment rather than _append_ to it (and NixOS relies on a heavily modified environment). Using the `env` command to append to the environment solves this issue.

I did initially try using

```lua
env = vim.tbl_extend("force", vim.loop.os_environ(), env or {})
```

to actually use the `env` spawn option, but that caused additional issues

### CODE REVIEW

**Comment by euclidianAce** [comment]
Fixes #20997

**Comment by lewis6991** [comment]
Make sure to add comments. The vim version used the `env` command too but I removed it for the env field.

File: runtime/lua/man.lua [hunk 1]
```diff
@@ -21,11 +21,21 @@ local function system(cmd, silent, env)
   local done = false
   local exit_code
 
+  -- we use the `env` command here rather than the env option to vim.loop.spawn since spawn will
+  -- completely overwrite the environment when we just want to modify the existing one
```
Comment (lewis6991) [inline runtime/lua/man.lua:25]: Replacing the environment is ok on the majority of systems. Probably need to mention we only need to do this for nix.

**Comment by justinmk** [comment]
> I did initially try using
> 
> ```lua
> env = vim.tbl_extend("force", vim.loop.os_environ(), env or {})
> ```

For reference, `jobstart(..., {env=...})` takes care of all of this. Ideally we would use jobstart(), or if we need synchronous behavior, ideally system() would have a `env` field. Or `jobwait(jobstart())` could return output.

File: runtime/lua/man.lua [hunk 2]
```diff
@@ -21,11 +21,23 @@ local function system(cmd, silent, env)
   local done = false
   local exit_code
 
+  -- We use the `env` command here rather than the env option to vim.loop.spawn since spawn will
+  -- completely overwrite the environment when we just want to modify the existing one.
+  --
+  -- Overwriting mainly causes problems NixOS which relies heavily on a non-standard environment.
+  local actual_cmd
+  if env then
+    actual_cmd = { "env" }
```
Comment (zeertzjq) [inline runtime/lua/man.lua:30]: Lint failure
```suggestion
    actual_cmd = { 'env' }
```

File: runtime/lua/man.lua [hunk 3]
```diff
@@ -21,11 +21,23 @@ local function system(cmd, silent, env)
   local done = false
   local exit_code
 
+  -- We use the `env` command here rather than the env option to vim.loop.spawn since spawn will
+  -- completely overwrite the environment when we just want to modify the existing one.
+  --
+  -- Overwriting mainly causes problems NixOS which relies heavily on a non-standard environment.
+  local actual_cmd
```
Comment (justinmk) [inline runtime/lua/man.lua:28]: rename the `cmd` param to `cmd_`, that avoids churn

**Comment by lewis6991** [comment]
> For reference, jobstart(..., {env=...}) takes care of all of this. Ideally we would use jobstart(), or if we need synchronous behavior, ideally system() would have a env field. Or jobwait(jobstart()) could return output.

I'd strongly prefer that we avoid using the vimscript bridge as much as we can, especially in perf sensitive code like this.

**Comment by justinmk** [comment]
ok, to unblock this, don't worry about using jobstart() for now. We'll revisit after https://github.com/neovim/neovim/pull/17620

**Review by lewis6991 — APPROVED** [review]

**Pull request merged**